### PR TITLE
Add IdleStateHandler after SslHandler

### PR DESCRIPTION
**Pull request by xinlian12**
- Enable idleStateHandler after sslHandshake has been completed so the idleState check will not happen before TLS handshake completed. 
- Previously, SDK is using the default sslHandshakeTimeout (10s), change to use connectionTimeout
- Capture inner sslHandshake related exception. 
    **Before the change**,  the exception message included in the diagnostics is `"exceptionMessage": "failed to establish connection to cdb-ms-prod-westus3-fd5.documents.azure.com:14355 due to java.nio.channels.ClosedChannelException",`
    **After the change**, the exception message included in diagnostics is `"exceptionMessage": "failed to establish connection to cdb-ms-prod-westus3-fd5.documents.azure.com:14355 due to io.netty.handler.ssl.SslHandshakeTimeoutException: handshake timed out after 10ms",`

- Capture RntbdContextException related exception
    **Before the change**,  the exception message included in the diagnostics is `"exceptionMessage": "failed to establish connection to cdb-ms-prod-westus3-fd5.documents.azure.com:14355 due to java.nio.channels.ClosedChannelException",`
    **After the change**, the exception message included in diagnostics is `"exceptionMessage": "failed to establish connection to cdb-ms-prod-westus3-fd5.documents.azure.com:14355 due to com.azure.cosmos.implementation.directconnectivity.rntbd.RntbdContextException: 500 Internal Server Error: null",`

Benchmark tests:
Document Count: 100,000, RU configured: 100,000
**Throughput:**
image

**P95 Latency:**
image

P99 Latency:
image

**P999 Latency:**
image

**Max Latency:**
image

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java [hunk 1]
```diff
@@ -378,6 +384,29 @@ public void userEventTriggered(final ChannelHandlerContext context, final Object
...
+                if (!sslHandshakeCompletionEvent.isSuccess()) {
+                    if (logger.isDebugEnabled()) {
+                        logger.debug("SslHandshake failed", sslHandshakeCompletionEvent.cause());
+                    }
+                    this.exceptionCaught(context, sslHandshakeCompletionEvent.cause());
+                    return;
+                } else {
+                    logger.info("adding idleStateHandler");
+                    context.pipeline().addFirst(
+                            new IdleStateHandler(
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdRequestManager.java:402]: indent

**Review by FabianMeiswinkel — APPROVED** [review]
Nice catch(es)! LGTM!

**Comment by xinlian12** [comment]
Known failed tests. Will have a different PR to fix the flaky tests

**Pull request merged**